core/src/main/java/hudson/model/Label.java

```diff
@@ -204,7 +215,7 @@ public boolean isSelfLabel() {
         Set<Node> nodes = this.nodes;
         if(nodes!=null) return nodes;
 
-        Set<Node> r = new HashSet<Node>();
+        Set<Node> r = new TreeSet<Node>(new NodeSorter());
```

Comment:
@daniel-beck thank you for adding me

I'm not sure that this is the best way to achieve your goal - from my point of view this kind of sorting would be much better to do on UI side.
In the current implementation TreeSet will be re-balanced every time we add a new node - it would leat to hundreds of re-allocations.
And because `getNodes` is used in a tons of other methods (only in Jenkins Core), I would prefer to not switch to TreeSet, but maybe expose another method that will be used by UI only (if we want to have this logic on server side).

Reply:
Fixed